includes/sanitizers/class-amp-core-theme-sanitizer.php

```diff
@@ -1932,6 +1998,257 @@ public function add_twentytwenty_toggles() {
+		$state_id     = str_replace( '-', '_', $toggle_class );
+
+		$body_id     = $this->dom->getElementId( $this->dom->body );
+		$document_id = $this->dom->getElementId( $this->dom->documentElement );
+
+		AMP_DOM_Utils::add_amp_action( $button, 'tap', "AMP.setState({{$state_id}: !{$state_id}})" );
+		AMP_DOM_Utils::add_amp_action( $button, 'tap', "{$body_id}.toggleClass(class='{$toggle_class}')" );
+		AMP_DOM_Utils::add_amp_action( $button, 'tap', "{$document_id}.toggleClass(class='{$toggle_class}')" );
+
+		$button->setAttribute( 'data-amp-bind-aria-pressed', "{$state_id} ? 'true' : 'false'" );
+	}
+
+	/**
+	 * Make the mobile menu for the Twenty Twenty-One theme AMP compatible.
+	 */
+	public function add_twentytwentyone_mobile_modal() {
+		$menu_query  = $this->dom->xpath->query( "//div[ @class and contains( concat( ' ', normalize-space( @class ), ' ' ), ' primary-menu-container ' ) ]" );
+		$menu_toggle = $this->dom->getElementById( 'primary-mobile-menu' );
+
+		if ( 1 !== $menu_query->length || ! $menu_toggle ) {
+			return;
+		}
+
+		/** @var DOMElement $primary_menu */
+		$primary_menu      = $menu_query->item( 0 );
+		$primary_menu_copy = $primary_menu->cloneNode( true );
+		foreach ( $this->dom->xpath->query( './/*[ @id ]', $primary_menu_copy ) as $element ) {
+			/** @var DOMElement $element */
+			$element->setAttribute( 'id', $element->getAttribute( 'id' ) . '-mobile' );
+		}
+
+		$body_id = $this->dom->getElementId( $this->dom->body, 'body' );
+
+		// Create an <amp-lightbox> element that will contain the modal.
+		$amp_lightbox = $this->dom->createElement( 'amp-lightbox' );
+		$amp_lightbox->setAttribute( 'layout', 'nodisplay' );
+		$amp_lightbox->setAttribute( 'animate-in', 'fade-in' );
+		$amp_lightbox->setAttribute( 'scrollable', true );
+		$amp_lightbox_id = $this->dom->getElementId( $amp_lightbox );
+
+		$state_string = str_replace( '-', '_', $amp_lightbox_id );
+
+		AMP_DOM_Utils::add_amp_action( $menu_toggle, 'tap', "{$amp_lightbox_id}.open" );
+		AMP_DOM_Utils::add_amp_action( $menu_toggle, 'tap', "{$body_id}.toggleClass(class=primary-navigation-open,force=true)" );
+
+		AMP_DOM_Utils::add_amp_action( $amp_lightbox, 'lightboxOpen', "AMP.setState({{$state_string}:true})" );
+		AMP_DOM_Utils::add_amp_action( $amp_lightbox, 'lightboxClose', "AMP.setState({{$state_string}:false})" );
+		AMP_DOM_Utils::add_amp_action( $amp_lightbox, 'lightboxClose', "{$body_id}.toggleClass(class=primary-navigation-open,force=false)" );
+
+		$menu_toggle->setAttribute( 'data-amp-bind-aria-expanded', "{$state_string} ? 'true' : 'false'" );
+
+		$amp_lightbox->appendChild( $primary_menu_copy );
+		$primary_menu->parentNode->insertBefore( $amp_lightbox, $primary_menu );
+	}
+
+	/**
+	 * Make the sub-menu functionality for the Twenty Twenty-One theme AMP compatible.
+	 *
+	 * Note: Hover functionality is accomplished through CSS.
+	 *
+	 * @see amend_twentytwentyone_styles()
+	 */
+	public function add_twentytwentyone_sub_menu_fix() {
```

Comment:
@pierlon (I might be placing this comment in an irrelevant place.) In AMP Standard mode, the subnav background colors aren't using my custom color:

<img width="590" alt="Screen Shot 2020-12-08 at 3 36 23 PM" src="https://user-images.githubusercontent.com/9020968/101544450-52e5ae00-396b-11eb-9988-36722e40e0de.png">

Reply:
Hmm I can only replicate that in the Customizer, and that seems to be an issue with the theme and not the plugin.

Reply:
If I refresh after publishing the changes the sub-menu color changes as expected.

Reply:
Not sure what I'm doing differently, but I can't seem to make this issue go away. See here: https://github.com/ampproject/amp-wp/pull/5648/files#r538847190

Reply:
I'm seeing this issue on the frontend as well. I changed the background color to red, and in the non-AMP version I got:

![image](https://user-images.githubusercontent.com/134745/101556079-dc4aaf80-396e-11eb-90d8-70184e47aad1.png)

But in AMP I get:

![image](https://user-images.githubusercontent.com/134745/101556109-ecfb2580-396e-11eb-9e3c-fc81302ce926.png)

Reply:
Looks like the `twenty-twenty-one-style-inline-css` styles are not being enqueued in AMP. Investigating why...

Reply:
In the non-AMP version there is this style rule added:

```html
<style id='twenty-twenty-one-style-inline-css'>
:root{--global--color-background: #ffeaea;--global--color-primary: #000;--global--color-secondary: #000;--button--color-background: #000;--button--color-text-hover: #000;}
</style>
```

In the `amp-custom` I am seeing:

```css
:root{--global--color-background:#ffeaea;--global--color-primary:#000;--global--color-secondary:#000;--button--color-background:#000;--button--color-text-hover:#000}
```

And immediately following this rule is this ruke:

```css
:root {
    /*...*/
    --global--color-background: var(--global--color-green);
}
```

So it is getting overridden.

Reply:
I think the issue is that `\AMP_Core_Theme_Sanitizer::amend_twentytwentyone_styles()` is running _after_ `\Twenty_Twenty_One_Custom_Colors::editor_custom_color_variables()`, and this is causing the order order to be reversed due to the main stylesheet being moved to be an inline style.

Reply:
I think I got it…

Reply:
See d504d3b51.

Reply:
Works perfectly :ok_hand:, thanks.